Remotion/Data/DomainObjects.UnitTests/Mapping/InterfaceDefinitionTest.cs

```diff
@@ -0,0 +1,329 @@
+    {
+    }
+
+    private UnitTestStorageProviderStubDefinition _storageProviderDefinition;
+
+    private InterfaceDefinition _domainBaseInterface;
+    private InterfaceDefinition _personInterface;
+    private InterfaceDefinition _customerInterface;
+    private InterfaceDefinition _organizationalUnitInterface;
+
+    [SetUp]
+    public void SetUp ()
+    {
+      _storageProviderDefinition = new UnitTestStorageProviderStubDefinition("DefaultStorageProvider");
+
+      _domainBaseInterface = InterfaceDefinitionObjectMother.CreateInterfaceDefinitionWithDefaultProperties(typeof(IDomainBase));
+      _personInterface = InterfaceDefinitionObjectMother.CreateInterfaceDefinitionWithDefaultProperties(typeof(IPerson), extendedInterfaces: new[] { _domainBaseInterface });
+      _customerInterface = InterfaceDefinitionObjectMother.CreateInterfaceDefinitionWithDefaultProperties(typeof(ICustomer), extendedInterfaces: new[] { _personInterface });
+      _organizationalUnitInterface = InterfaceDefinitionObjectMother.CreateInterfaceDefinitionWithDefaultProperties(
+          typeof(IOrganizationalUnit),
+          extendedInterfaces: new[] { _domainBaseInterface });
+
+      _domainBaseInterface.SetExtendingInterfaces(new[] { _personInterface, _organizationalUnitInterface });
+      _personInterface.SetExtendingInterfaces(new[] { _customerInterface });
+    }
+
+    [Test]
+    public void Initialize ()
+    {
+      var actual = new InterfaceDefinition(
+          typeof(IDomainBase),
+          Array.Empty<InterfaceDefinition>(),
+          null,
+          DefaultStorageClass.Transaction);
+
+      Assert.That(actual.Type, Is.SameAs(typeof(IDomainBase)));
+      Assert.That(actual.ExtendedInterfaces, Is.Empty);
+      Assert.That(actual.DefaultStorageClass, Is.EqualTo(DefaultStorageClass.Transaction));
+
+      Assert.That(actual.HasStorageEntityDefinitionBeenSet, Is.False);
+      Assert.That(
+          () => actual.StorageEntityDefinition,
+          Throws.InvalidOperationException
+              .With.Message.EqualTo(
+                  "StorageEntityDefinition has not been set for type 'Remotion.Data.DomainObjects.UnitTests.Mapping.InterfaceDefinitionTest+IDomainBase'."));
+
+      Assert.That(
+          () => actual.ImplementingClasses,
+          Throws.InvalidOperationException
+              .With.Message.EqualTo(
+                  "No implementing classes have been set for interface 'Remotion.Data.DomainObjects.UnitTests.Mapping.InterfaceDefinitionTest+IDomainBase'."));
+      Assert.That(
+          () => actual.ExtendingInterfaces,
+          Throws.InvalidOperationException
+              .With.Message.EqualTo(
+                  "No extending interfaces have been set for interface 'Remotion.Data.DomainObjects.UnitTests.Mapping.InterfaceDefinitionTest+IDomainBase'."));
+
+      Assert.That(actual.IsReadOnly, Is.False);
+    }
+
+    [Test]
+    public void IsPartOfInheritanceHierarchy ()
```

Comment:
Am realizing that a property IsPartOfInheritanceHierarchy is very strange. How is this property used?

Reply:
It is used to determine if a class ID column if necessary for a relation

Reply:
Hmm... Move to TypeDefinitionHierarchy because it has a utility feel to it instead of a first-class citizien on the type?

Reply:
or just a private logic in the persistence layer when doing the classID column?